src/url.js

```diff
@@ -123,6 +123,15 @@ export function parseUrl(url) {
 }
 
 /**
+ * Duplicate of parseUrl, but return a cloned version of the returned obj.
+ * @param {string} url
+ * @return {!Location}
+ */
+export function parseUrlAndClone(url) {
```

Comment:
Added to presubmit check. Thought I think this only provides a convenient way to clone a `Location` object. Even we restrict this, caller can also achieve it themselves.